### ISSUE TITLE
The date for the RM spec is stuck

### ISSUE THREAD

**Issue by berezovskyi** [oslc-op/oslc-specs#61]
When I navigate to https://open-services.net/spec/core/v3.0-psd-20190620, I expect to see 2019-06-20 as the date on the page (commit date). The date is stuck at 31 May 2018.

**Comment by jamsden** [comment]
The May 2018 date is correct. The date displayed is the publishDate - the date that revision of the document was published. This will be updated when we publish the PS (maybe for the PSD, I'm not sure). I think its only the date of a revision published in the OASIS Library as that is where the document lifecycle is publicly manifest.

**Comment by berezovskyi** [comment]
No, this is a working draft in the master branch. The date on it should correspond to the date it was last edited.

**Comment by berezovskyi** [comment]
If we can’t fix it to show the right date, we need to remove the date from working drafts altogether.

--
/Andrew
(from phone)



On Thu, Jun 27, 2019 at 10:39 PM +0200, "Jim Amsden" <notifications@github.com<mailto:notifications@github.com>> wrote:


The May 2018 date is correct. The date displayed is the publishDate - the date that revision of the document was published. This will be updated when we publish the PS (maybe for the PSD, I'm not sure). I think its only the date of a revision published in the OASIS Library as that is where the document lifecycle is publicly manifest.

**Comment by jamsden** [comment]
I would prefer that the date stays the last publish date. The date last edited is available from GitHub and doesn't need to be redundantly captured in the document itself. But the publish data is an important part of the lifecycle of the document, reflecting when the current status and revision was voted on.

**Comment by berezovskyi** [comment]
No, you can't see the GH date when you view the spec draft through Githack. And it confuses me a lot!


--
/Andrew
________________________________

**Comment by ndjc** [comment]
Here is what ReSpec does today:
publishDate: the date to use for the publication, default to document.lastModified, and failing that to now. The format is YYYY-MM-DD or a Date object.
We can add another field to the templates for the last modified date regardless of publication date if you want.

**Comment by berezovskyi** [comment]
Is there a way to suppress the output of the date for the WD status, Nick?

--
/Andrew
(from phone)



On Fri, Jun 28, 2019 at 7:21 PM +0200, "Nick Crossley" <notifications@github.com<mailto:notifications@github.com>> wrote:


Here is what ReSpec does today:
publishDate: the date to use for the publication, default to document.lastModified, and failing that to now. The format is YYYY-MM-DD or a Date object.
We can add another field to the templates for the last modified date regardless of publication date if you want.

**Comment by berezovskyi** [comment]
All this makes me think we need a CI pipeline for this, unfortunately.

--
/Andrew
(from phone)



On Fri, Jun 28, 2019 at 7:21 PM +0200, "Nick Crossley" <notifications@github.com<mailto:notifications@github.com>> wrote:


Here is what ReSpec does today:
publishDate: the date to use for the publication, default to document.lastModified, and failing that to now. The format is YYYY-MM-DD or a Date object.
We can add another field to the templates for the last modified date regardless of publication date if you want.

**Comment by ndjc** [comment]
Either suppressing the date for a WD, or changing it to use the document modify date, would be easy. ReSpec is very aware of the doc status.
However, this may not be necessary if we use the techniques I described in https://github.com/oslc-op/oslc-specs/issues/62#issuecomment-506819786.

**Comment by berezovskyi** [comment]
I still want to have a good date that reflects reality. I was about to give up on this but then saw https://shexspec.github.io/spec/ 🤷‍♂

**Comment by ndjc** [comment]
I'm happy to change ReSpec in any way that meets the team's consensus. Let's discuss at the next meeting which I attend. I'm out July 4 and July 11, so that would be July 18.

**Comment by berezovskyi** [comment]
I think I finally cracked this (and the publication process). To fix it in the best way possible would require a breaking change, but the effort is relatively minor.

So, I almost gave up on editing the source to produce the right dates and was about to go all in with a CI pipeline when I found out that ReSpec does not have a good CLI. Not until version 20.10, at least.

But ShEx somehow showed a reasonable draft modification date (2 weeks ago, matches the commit) without stating it explicitly. From the knowledge @ndjc equipped me earlier, I started investigating the `document.lastModified` property returned by Github Pages and GitHack. Turns out GH Pages return a truthful value, while GitHack always returns `now()`.

The obvious next step was to figure out to switch to GH Pages. I then realised that ShEx pushes straight to the `gh-pages` branch. I branched off our `master` and pushed and I saw the spec and the modification date corresponded to the time of push! 🎉 However, there were 2 problems: `gh-pages` is an ugly branch and the timestamps for _all_ specs get updated at the same time. For the former, I just configured the repo to publish the pages straight from the master branch.

But the other problem is not easily solvable:

1. we can still have a CI script that will deploy the `/specs` folder on e.g. `specs.open-services.net` or
2. we split all specs into individual repos and then the update to one spec will not disturb the last modification date of the other specs (I prefer this approach, plus version tags and issues will be grouped by the spec; we can let OASIS publish the PSDs).
3. Or option 2 for the WDs and option 1 for the PSDs.

I just updated the dates and statuses on the RM spec and tomorrow you will see that the modification for it is still July 1st: https://oslc-op.github.io/oslc-specs/specs/rm/requirements-management-spec.html

We went full circle to GH Pages as @jamsden was initially suggesting 🤸‍♂
